src/main/xerus/monstercat/api/Covers.kt

```diff
@@ -5,26 +5,35 @@ import org.apache.http.HttpEntity
 import org.apache.http.client.methods.HttpGet
 import xerus.ktutil.replaceIllegalFileChars
 import xerus.monstercat.cacheDir
+import java.io.File
 import java.io.InputStream
 import java.net.URI
 
 object Covers {
 	
-	private val coverCacheDir = cacheDir.resolve("cover-images").apply { mkdirs() }
+	private val thumbnailCacheDir = cacheDir.resolve("cover-images").apply { mkdirs() }
+	private val coverCacheDir = cacheDir.resolve("large-cover-images").apply { mkdirs() }
 	
-	/** Returns an Image of the cover in the requested size using caching.
-	 * @param size the size of the Image - the underlying image data will always be 64x64, thus this is the default. */
-	fun getCoverImage(coverUrl: String, size: Number = 64, invalidate: Boolean = false): Image =
-		getCover(coverUrl, invalidate).use { createImage(it, size) }
+	private fun coverCacheFile(file: File, coverUrl: String) =
+			file.apply { mkdirs() }.resolve(coverUrl.substringAfterLast('/').replaceIllegalFileChars())
 	
 	private fun createImage(content: InputStream, size: Number) =
-		Image(content, size.toDouble(), size.toDouble(), false, false)
+			Image(content, size.toDouble(), size.toDouble(), false, false)
 	
-	/** Returns an InputStream to the cover in size 64x64, using caching. */
-	fun getCover(coverUrl: String, invalidate: Boolean = false): InputStream {
-		val file = coverCacheFile(coverUrl)
+	/** Returns an Image of the cover in the requested size using caching.
+	 * @param size the size of the Image - the underlying image data will always be 64x64, thus this is the default. */
+	fun getThumbnailImage(coverUrl: String, size: Number = 64, invalidate: Boolean = false): Image =
+		getCover(coverUrl, 64,  invalidate).use { createImage(it, size) }
+
+	/** @return an [InputStream] of the downloaded image file
+	 * @param coverUrl the URL for fetching the cover
+	 * @param size the final size of the cover
+	 * @param invalidate set to true to ignore already existing cache files
+	 */
+	fun getCover(coverUrl: String, size: Int = 64, invalidate: Boolean = false) : InputStream {
+		val file = coverCacheFile(if (size > 128) coverCacheDir else thumbnailCacheDir, coverUrl)
```

Comment:
I'd just use the same directory and then pass the size to this method so it is appended to the filename, i.e. "coverx64.jpg"

Reply:
Have you read my answer above ?

Reply:
oh, it got lost. Then we'll leave it like that for now, maybe I reconsider this in the future.

Reply:
I actually implemented it. Up to you to revert it or not. adc75cd97b5ecc7555fab0df8d66264266e00984